Index: python-vobject-154/setup.py
===================================================================
--- python-vobject-154.orig/setup.py	2006-09-06 11:16:58.000000000 +0200
+++ python-vobject-154/setup.py	2006-09-06 11:17:14.000000000 +0200
@@ -6,10 +6,6 @@
 Requires dateutil (https://moin.conectiva.com.br/DateUtil) 0.9 or later.
 """
 
-# not using setuptools until Chandler's ready for eggs
-from ez_setup import use_setuptools
-use_setuptools()
-
 from setuptools import setup, find_packages
 
 #from distutils.core import setup
