Index: python-vobject/setup.py
===================================================================
--- python-vobject.orig/setup.py	2007-06-18 18:09:28.000000000 +0200
+++ python-vobject/setup.py	2007-06-18 18:09:43.000000000 +0200
@@ -38,7 +38,6 @@
       license = "Apache",
       zip_safe = True,
       url = "http://vobject.skyhouseconsulting.com",
-      entry_points = { 'console_scripts': ['ics_diff = vobject.ics_diff:main'] },
       package_dir = {'':'src'},
       package_data = {'': ALL_EXTS},
 
