diff --git a/setup.py b/setup.py
index 1c70429..d2d0ee1 100755
--- a/setup.py
+++ b/setup.py
@@ -48,8 +48,6 @@ setup(name = "vobject",
       license = "Apache",
       zip_safe = True,
       url = "http://vobject.skyhouseconsulting.com",
-      entry_points = { 'console_scripts': ['ics_diff = vobject.ics_diff:main',
-                                           'change_tz = vobject.change_tz:main']},
       include_package_data = True,
       test_suite = "test_vobject",
 
